app/services/work_uploads_edit_service.rb

```diff
@@ -11,9 +11,12 @@ def initialize(work, current_user)
   def update_precurated_file_list(added_files, deleted_files)
     delete_uploads(deleted_files)
     add_uploads(added_files)
-    if work.changes.count > 0
+    file_list_changed = deleted_files.count > 0 || added_files.count > 0
+    if file_list_changed
       s3_service.client_s3_files(reload: true)
-      work.reload # reload the work to pick up the changes in the attachments
+      # Pick up the changes in the attachments and assings them to the current user
+      work.reload
+      work.reload_snapshots(user_id: current_user.id)
```

Comment:
This is the big change on this PR: I am forcing to reload the snapshots so that the changes are assigned to the current user.